### PR TITLE
[FLINK-30456] Fixing Version and provider in OLM Description

### PR DESCRIPTION
**Pull request by jbusche**
Signed-off-by: James Busche <jbusche@us.ibm.com>

## What is the purpose of the change

Fixes OLM Documentation issues that I noticed 
- basic.yaml was described using an old version
- The labels were hardcoded to 1.2.0
- The empty Provider needs to change to something to avoid future operator-sdk CI issues.  Changing to "Community" for now but possibly can make "The Apache Software Foundation" if people agree.

## Brief change log

Fix basic.yaml version in:
- generate-olm-bundle.sh
- docker-entry.sh
Fix label version issues in:
- docker-entry.sh
Fix Provider in the csv-template
- csv-template/bases/flink-kubernetes-operator.clusterserviceversion.yaml

## Verifying this change
git clone https://github.com/jbusche/flink-kubernetes-operator.git

git checkout FLINK-30456

cd /tools/olm

vi env.sh and replace:
```
OPERATOR_IMG=ghcr.io/apache/flink-kubernetes-operator:main
with
OPERATOR_IMG=docker.io/apache/flink-kubernetes-operator:1.3.0
```
./generate-olm-bundle.sh

Follow the output provided from the ./generate-olm-bundle.sh command to create a catsrc and subscription.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: (no)
  - Core observer or reconciler logic that is regularly executed: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by morhidi** [comment]
@jbusche please fill out the PR template properly

**Comment by jbusche** [comment]
Yes @morhidi it took me awhile to fill out completely

**Comment by jbusche** [comment]
I tested my PR briefly on OpenShift 4.10, will do some more thorough testing tomorrow and post the results here.  I think @tedhtchang might test on his server(s) too.

File: tools/olm/docker-entry.sh [hunk 1]
```diff
@@ -102,6 +96,10 @@ generate_olm_bundle() {
   yq ea -i ".spec.replaces = \"${PACKAGE_NAME}.v${PREVIOUS_BUNDLE_VERSION}\" | .spec.replaces style=\"\"" "${CSV_FILE}"
 
   yq ea -i "del(.subjects[0].namespace)" "${ROLE_BINDING}"
+
+  # Needed to replace description with new bundle values
+  sed -i "s/release-1.1/release-${BUNDLE_VERSION}/" "${CSV_FILE}"
+  sed -i "s/version: 1.2.0/version: ${BUNDLE_VERSION}/" "${CSV_FILE}"
```
Comment (gyfora) [inline tools/olm/docker-entry.sh:102]: would it make sense to add some nicer placeholder strings in the CSV_FILE ?

Comment (jbusche) [reply tools/olm/docker-entry.sh]: That's a great idea @gyfora, I've added a RELEASE_VERSION placeholder and also I've added a few items under links, specifically:
    - name: Website
      url: https://flink.apache.org/
    - name: Documentation
      url: https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/
    - name: Mailing list
      url: https://lists.apache.org/list.html?dev@flink.apache.org
    - name: Slack
      url: https://apache-flink.slack.com/join/shared_invite/zt-1llkzbgyt-K2nNGGg88rfsDGLkT09Qzg#/shared-invite/email
    - name: GitHub
      url: https://github.com/apache/flink-kubernetes-operator

**Comment by jbusche** [comment]
OK, let me test this one on my OpenShift cluster, will post results here

**Comment by jbusche** [comment]
It's looking good - it added the links as expected, and is matching on the versions nicely.  
Deployed well:
```
oc get pods
NAME                                                              READY   STATUS      RESTARTS   AGE
0e60453bbf27f3b2271a54772da7acb269b807d208b10b221a80c740496p2ml   0/1     Completed   0          6m33s
basic-example-56876dc586-wwrbv                                    1/1     Running     0          4m50s
basic-example-taskmanager-1-1                                     1/1     Running     0          4m38s
flink-kubernetes-operator-7dd84c798f-52d4k                        2/2     Running     0          6m21s
olm-flink-operator-catalog-9s9ch                                  1/1     Running     0          7m8s
```

File: tools/olm/docker-entry.sh (hunk 1 above)
Comment (tedhtchang) [reply tools/olm/docker-entry.sh]: @jbusche Links look good!
image

**Comment by morhidi** [comment]
+1 LGTM

**Comment by tedhtchang** [comment]
Tested on Kubernetes. LGTM

**Pull request merged**